### PR TITLE
Initial tentative changes to displaying section.

### PR DESCRIPTION
**Pull request by rhit-tatebr**

### CODE REVIEW

File: packages/celer-web-app/src/ui/components/Map/Map.tsx [hunk 1]
```diff
@@ -37,35 +39,39 @@ const mapCanvas = new MapCanvas();
 const IconSize = 32;
 const IconSizeSmall = 24;
 
-const InGameOriginGeoCoord = inGameToGeoCoord(inGameCoord(0,0));
-const DynamicCanvasAnchorGeoCoord = inGameToGeoCoord(inGameCoord(-DynamicCanvasSizeX,-DynamicCanvasSizeZ));
+const InGameOriginGeoCoord = inGameToGeoCoord(inGameCoord(0, 0));
+const DynamicCanvasAnchorGeoCoord = inGameToGeoCoord(inGameCoord(-DynamicCanvasSizeX, -DynamicCanvasSizeZ));
 
-export const Map: React.FC<MapProps> = ({icons, lines, manualCenter}) => {
```
Comment (iTNTPiston) [inline packages/celer-web-app/src/ui/components/Map/Map.tsx:43]: The filtering logic for icons and lines should really be in MapFrame. Map is a simple component used by MapFrame and it should just render whatever is passed into it

**Review by iTNTPiston — CHANGES_REQUESTED** [review]

File: packages/celer-web-app/src/ui/components/Map/Map.tsx (hunk 1 above)
Comment (rhit-tatebr) [reply packages/celer-web-app/src/ui/components/Map/Map.tsx:43]: It has been moved. I added a usememo as well to make sure filtering the icons and lines wouldn't cause lag.

File: packages/celer-web-app/src/core/map/MapEngine.ts [hunk 2]
```diff
@@ -38,6 +42,10 @@ export class MapEngine {
 	}
 
 	private computeLine(line: DocLine, outIcons: MapIcon[], outLines: MapLine[]): void {
+		if(line.lineType === "DocLineSection"){
+			this.endCurrentPathIfNeed(outLines);
+			this.currentSection = line.sectionNumber;
+		}
 		if(line.lineType === "DocLineBanner" || line.lineType === "DocLineSection") {
```
Comment (iTNTPiston) [inline packages/celer-web-app/src/core/map/MapEngine.ts:49]: The second half of this `if` is already covered above

**Review by iTNTPiston — CHANGES_REQUESTED** [review]
Just one small thing. The rest looks good

File: packages/celer-web-app/src/core/map/MapEngine.ts (hunk 2 above)
Comment (rhit-tatebr) [reply packages/celer-web-app/src/core/map/MapEngine.ts]: Alright, I moved it around. I'm not entirely sure if that's what you meant though.

Comment (iTNTPiston) [reply packages/celer-web-app/src/core/map/MapEngine.ts]: I meant something like this
```
if (line.lineType === "DocLineBanner") {
return;
}

if (line.lineType === "DocLineSection") {
this.endCurrentPathIfNeed(outLines);
this.currentSection = line.sectionNumber;
return;
}
```
Then you don't have repeated conditions in the ifs

**Review by iTNTPiston — APPROVED** [review]

**Pull request merged**